### DISCUSSION TITLE
Add ability to mix and match kgs and lbs

### DISCUSSION THREAD

**Discussion by astashov** [astashov/liftosaur#134] [Ideas]
In some gyms, there could be machines with weight in lbs, and plates with the weight in kgs. Also, powerlifting competitions mostly use kgs, even in US.

Need to figure out a way to mix and match kgs and lbs in the same program/workout.

**Comment by yrinkel** [comment]
For an extension of that idea, see the Plate pools / sharing of plates idea:
https://github.com/astashov/liftosaur/discussions/143#discussion-7097305

**Comment by yrinkel** [comment]
As for implementation:
- each equipment has a default unit, and the user can change the unit in the equipment’s configuration. (Lbs/Kg). By default the equipment will be set to the global unit from the user’s preferences. So equipment is the source of truth regarding unit.

- Internally Liftosaur can use *only one unit*, a very small unit so as to not create rounding errors in weight storage and calculations (for example grams). 

- each weight configured is immediately converted to that small unit (grams) internally for storage and calculations, and only ever converted back to the equipment’s configured unit *when displayed on the screen*. 

- graphs, history, volume, etc… are shown in the unit selected by the user in the general settings. That’s the unit the user is most comfortable with. 

Advantages of this approach:
- it correctly handles totally mixed equipment (ex: a barbell in Kg, loaded with plates in Lbs). 
- it correctly handles mixed cases such as equipment configured in Kg but Liftoscript progress increments coded in Lbs. 
- it should not have rounding issues (1 gram is 1/1,000th of a Kilogram and 0.0022 lbs). 
- it will automatically adjust if the user changes the default unit of the program or the set unit of an equipment (including in the history). 
- it shouldn’t require too much code change, as you are probably using only one unit internally already. So changes will mostly limited to renaming the internal unit, option for setting per equipment unit, and handling input/output conversion between the equipment’s unit and the internal unit (pretty straightforward).

**Reply by astashov** [reply]
This is done!
